src/app/GameController.ts

```diff
@@ -1,65 +1,77 @@
-// import { time } from 'console';
 import { AnimalNames } from '../Enums/AnimalNamesEnum';
-import { Player } from '../Player';
-import { BreedProcessor } from './BreedProcessor';
-import { Bank } from './logic/Bank';
-import { Timer } from './Timer';
+import { GameProcessor } from './logic/GameProcessor';
 import { View } from './View';
+import { Game } from './logic/Game';
+import { defaultGameConfiguration } from './logic/defaultGameConfiguration';
 
 export class GameController {
-  private currentPlayer: Player | undefined;
-  private player: Player | undefined;
-  private timer: Timer;
-  private breedProcessor: BreedProcessor;
+  private game: Game;
+  private gameProcessor: GameProcessor;
   constructor(private view: View) {
-    this.timer = new Timer();
-    this.breedProcessor = new BreedProcessor(new Bank());
+    this.game = new Game(defaultGameConfiguration);
+    this.gameProcessor = new GameProcessor(this.game);
   }
 
-  get theCurrentPlayer(): Player {
-    return this.currentPlayer as Player;
+  get theGame(): Game {
+    return this.game;
   }
 
-  get theTimer(): Timer {
-    return this.timer;
+  /**
+   * Starts the turn for the current player.
+   * Displays alert when the time is over.
+   * Updates the remaining time on the View.
+   */
+  startTurn(): void {
+    this.gameProcessor.startTurn(
+      (currentPlayer) => {
```

Comment:
Is it necessary to use callbacks here?

Reply:
Is there anything wrong with this code? Do you have any specific alternatives? 
By using those callbacks I wanted to avoid using View in GameProcessor.